docs/docs/react-query/queries.md

```diff
@@ -0,0 +1,47 @@
+---
+sidebar_position: 2
+---
+
+# Queries
+
+For every `get` endpoint in your API, there will be a `use<Action>()` method on the Stainless
+React Query client that wraps [`useQuery`](https://tanstack.com/query/v4/docs/react/reference/useQuery).
+
+For example, if `posts.retrieve` is a `get /api/posts/{postId}` endpoint, then
+`client.posts.useRetrieve(postId, [query], [reactQueryOptions])` will be available:
```

Comment:
Why's `query` paired with `reactQueryOptions` in mutations, but not in queries?

Reply:
Because in mutations you typically have body parameters in an object.  If you allowed a signature like `mutate([path], [query], [body], [options])`, that's potentially 1-3 objects, but each one of them is optional.  So how to tell what's what?

You could try to see if the last argument is `[mutationOptions]` by looking at its properties, for example if it has `onError`, `onSuccess`, etc.  And then if there's one other object it's `body`, if there's two other objects they're `query` and `body`.

But what if a new version of `react-query` adds some new option `foo`, and the last object is `{ foo: 1 }`?  Then our code would mistakenly conclude that's not `mutationOptions`, until we update our code.

We're already doing this kind of thing for queries, but I think it's gross:

```
// This is required so that we can determine if a given object matches the RequestOptions
// type at runtime. While this requires duplication, it is enforced by the TypeScript
// compiler such that any missing / extraneous keys will cause an error.
const useQueryOptionsKeys: KeysEnum<{ query?: any } & UseQueryOptions> = {
  context: true,
  retry: true,
  retryDelay: true,
  networkMode: true,
  cacheTime: true,
  isDataEqual: true,
  queryHash: true,
  queryKeyHashFn: true,
  initialData: true,
  initialDataUpdatedAt: true,
  behavior: true,
  structuralSharing: true,
  getPreviousPageParam: true,
  getNextPageParam: true,
  _defaulted: true,
  meta: true,
  query: true,
  enabled: true,
  staleTime: true,
  refetchInterval: true,
  refetchIntervalInBackground: true,
  refetchOnWindowFocus: true,
  refetchOnReconnect: true,
  refetchOnMount: true,
  retryOnMount: true,
  notifyOnChangeProps: true,
  onSuccess: true,
  onError: true,
  onSettled: true,
  useErrorBoundary: true,
  select: true,
  suspense: true,
  keepPreviousData: true,
  placeholderData: true,
  _optimisticResults: true,
};

export const isUseQueryOptions = (obj: unknown): obj is BaseUseQueryOptions => {
  return (
    typeof obj === "object" &&
    obj !== null &&
    !isEmpty(obj) &&
    Object.keys(obj).every((k) => Object.hasOwn(useQueryOptionsKeys, k))
  );
};
```

This is inspired by what we do for `isRequestOptions` in the base Stl client.  But in that case we control the properties on `RequestOptions` so we don't have the problem that we would if `react-query` adds new options.

Reply:
Actually I forgot, I do have to do a runtime `isMutateOptions` check.  It's certainly not as bad as the `isUseQueryOptions` check but I wish we could have something cleaner and more certain.  But I guess I could support top-level query in mutations with a bit more work...

Reply:
Should we require you pass an argument for body/query/whatever if you wanna pass mutation options, even if it's just `null`?

Reply:
Thinking about this more, maybe with the type info I can detect if there would be a conflict and choose a method signature type that ensures everything can be interpreted correctly

Reply:
maybe… the inconsistency could be an issue… how bad would it be to require the intermediary args? Do people pass mutationOptions a lot?